### PR TITLE
Feat: support specified existing namespace while creating a project

### LINKED ISSUE #2914: [Feature] 支持预定义的命名空间，作为apiserver的 Project/Target 命名空间
**Describe the solution you'd like**

因项目需要， 管理应用所在的命名空间是预先定义好的，命名规范、命名空间的个数，不能随便变动的。
为了使用velaux 管理应用，希望 vela-core apisever 支持预定义的命名空间。

kubevela目前的实现 pkg/apiserver/rest/usercase/project.go `projectUsecaseImpl.CreateProject` 
在添加项目时，是固定命名空间的格式（带前缀`project-`）: 

```golang
	new := &model.Project{
		Name:        req.Name,
                 ....
		Namespace:   fmt.Sprintf("project-%s", req.Name),
	}
```

并且在创建 Namespace k8s资源之后，才能录入持久化存储 ( `p.ds.Add(ctx, new)` ) ，所以已有的命名空间也不能被纳入管理

```golang 
	if err := p.kubeClient.Create(ctx, &corev1.Namespace{
		...
	}); err != nil {
		if !apierrors.IsAlreadyExists(err) {
			return nil, err
		}
	}

	if err := p.ds.Add(ctx, new); err != nil {
		return nil, err
	}
```

假设我已有的项目在 `local`，命名空间 `mygroup-qa` 里 已经部署了应用，
在此情况下，vela-core apiserver 不能纳管这里命名空间作为项目，也不能在后添加的addon velaux UI上进行管理。

### PR DESCRIPTION
**Pull request by barnettZQG**
Signed-off-by: barnettZQG <barnett.zqg@gmail.com>

### Description of your changes

support specified existing namespace while creating a project, but the namespace can not belong to other projects.

Fixes #2914

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/apiserver/rest/apis/v1/types.go [hunk 1]
```diff
@@ -483,6 +483,7 @@ type CreateProjectRequest struct {
 	Name        string `json:"name" validate:"checkname"`
 	Alias       string `json:"alias" validate:"checkalias" optional:"true"`
 	Description string `json:"description" optional:"true"`
+	Namespace   string `json:"namespace" optional:"true"`
```
Comment (wonderflow) [inline pkg/apiserver/rest/apis/v1/types.go:486]: This can be optional I think, if the namespace is empty, just use the project name.

Comment (barnettZQG) [reply pkg/apiserver/rest/apis/v1/types.go:486]: yes

File: pkg/apiserver/rest/usecase/project.go [hunk 2]
```diff
@@ -101,20 +102,39 @@ func (p *projectUsecaseImpl) CreateProject(ctx context.Context, req apisv1.Creat
...
+	if req.Namespace != "" {
+		new.Namespace = req.Namespace
+	}
 	// create namespace at first
 	if err := p.kubeClient.Create(ctx, &corev1.Namespace{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: new.Namespace,
 			Labels: map[string]string{
 				oam.LabelProjectNamesapce: new.Name,
+				oam.LabelUsageNamespace:   "project",
```
Comment (wonderflow) [inline pkg/apiserver/rest/usecase/project.go:114]: let's use a const variable for the `project`.

File: pkg/apiserver/rest/usecase/project.go [hunk 3]
```diff
@@ -101,20 +102,39 @@ func (p *projectUsecaseImpl) CreateProject(ctx context.Context, req apisv1.Creat
...
 			return nil, err
 		}
+		if apierrors.IsAlreadyExists(err) {
+			var namespace corev1.Namespace
+			if err := p.kubeClient.Get(ctx, k8stypes.NamespacedName{Name: new.Namespace}, &namespace); err != nil {
+				return nil, bcode.ErrProjectNamespaceFail
+			}
+			if _, exist := namespace.Labels[oam.LabelProjectNamesapce]; !exist {
+				namespace.Labels[oam.LabelProjectNamesapce] = new.Name
+				namespace.Labels[oam.LabelUsageNamespace] = "project"
```
Comment (wonderflow) [inline pkg/apiserver/rest/usecase/project.go:129]: same here

**Review by wonderflow — APPROVED** [review]
LGTM after nit

**Review by yangsoon — APPROVED** [review]

**Pull request merged**